pdl_syntax.md

```diff
@@ -0,0 +1,843 @@
+
+/**
+ * A foo record
+ */
+record Foo {
+   field1: string
+   /**
+    * field2 is an optional field.
+    */
+   field2: optional string
+}
+```
+
+Record fields may have default values.
+
+For example:
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * A foo record
+ */
+record Foo {
+   field1: string
+   field2: string = "message"
+}
+```
+
+An optional field may have default value.
+
+For example: 
+```
+namespace com.linkedin.pegasus.generator.test.idl.records
+
+record WithOptionalPrimitiveDefault {
+  intWithDefault: optional int = 1
+}
+```
+
+### Inline Record
+
+A record can contain inline records.
+
+For example:
+```
+namespace com.linkedin.pegasus.generator.test.idl.records
+
+record WithInlineRecord {
+  inline: record InlineRecord {
+    value: int
+  }
+  inlineOptional: optional record InlineOptionalRecord {
+    value: string
+  }
+}
+```
+
+Default value can be set up for a field that defines the inline records.
+
+For example: 
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record WithInlineRecord {
+
+    inline: record InlineRecord {
+      value: int
+    } = { "value" : 1 }
+
+    inlineOptional: optional record InlineOptionalRecord {
+      value: string
+    } = { "value" : "default-value" }
+}
+```
+
+Inline records can aslo be union members.
+
+For example:
+```
+namespace com.linkedin.pegasus.generator.test.idl.unions
+
+record UnionWithInlineRecord {
+  value = union[
+
+    record InlineRecord {
+      value: optional int
+    },
+
+    record InlineRecord2 {}
+  ]
+}
+```
+
+### Doc Strings
+
+Types and fields may be documented using “doc strings”.
+
+For example: 
+```
+/**
+ * Doc strings may be added to types. This doc should describe the purposes
+ * of the Example type.
+ */
+record Example {
+  /**
+   * Doc strings may also be added to fields.
+   */
+   field1: string
+
+   /** Doc strings can be single line.*/
+   field2: int
+}
+```
+
+**Note:**
+If you use Java comment style for doc string, e.g "// Doc String", those doc strings will not be stored in in-memory schema.
+
+### Including fields
+
+Record can include fields from one or more other records..
+
+For example: 
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * Bar includes fields of Foo, Bar will have fields b1 from itself and f1 from Foo
+ */
+record Bar includes Foo {
+  b1: string
+}
+```
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record Foo {
+  f1: string
+}
+```
+
+Multiple includes example:
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * Bar includes fields of Foo and Simple, Bar will have fields b1 from itself, f1 from Foo and s1 from Simple
+ */
+record Bar includes Foo, Simple {
+  b1: string
+}
+```
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record Simple {
+  s1: string
+}
+```
+
+In pegasus, field inclusion does not imply inheritance, it is merely a convenience to reduce duplication when writing schemas.
+
+
+### Properties
```

Comment:
As I mentioned offline, make this its own top-level section to describe the mechanics of properties. Then also add blurbs about adding properties to individual sections (e.g. enum symbols, record fields, aliased union members, etc.)

Reply:
fixed